.github/workflows/dockerpublish.yml

```diff
@@ -22,87 +22,63 @@ env:
 jobs:
   push:
     runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        registry:
+          - docker
+          - github
+
     steps:
       - uses: actions/checkout@v2
 
       - name: Build image
-        run: docker build --pull -f ./.docker/Dockerfile -t "${IMAGE_NAME}" ./.docker
+        run: docker build --pull -f ./.docker/Dockerfile -t "${{ env.IMAGE_NAME }}" ./.docker
 
-      - name: Log into docker hub registry
+      - name: Log into Docker Hub
+        if: matrix.registry == 'docker' && (startsWith(github.ref, 'refs/heads/master') || startsWith(github.ref, 'refs/heads/tags'))
```

Comment:
@localheinz Just FYI: I needed to fix the check for tags here:
```suggest
        if: matrix.registry == 'docker' && (startsWith(github.ref, 'refs/heads/master') || startsWith(github.ref, 'refs/tags'))
```

Now everything works as expected, see: 
* https://github.com/hollodotme/readis-docker-image/actions/runs/395633404
* https://github.com/hollodotme/readis-docker-image/actions/runs/395633402

🚀